cmd/zdb/zdb.c

```diff
@@ -4574,12 +4579,75 @@ dump_path(char *ds, char *path)
 
 	(void) snprintf(curpath, sizeof (curpath), "dataset=%s path=/", ds);
 
-	err = dump_path_impl(os, root_obj, path);
+	err = dump_path_impl(os, root_obj, path, retobj);
 
 	close_objset(os, FTAG);
 	return (err);
 }
 
+static int
+zdb_copy_object(objset_t *os, uint64_t srcobj, char *destfile)
+{
+	int err = 0;
+	int fd;
+	char *buf = NULL;
+	uint64_t size, readsize, oursize, offset;
+	sa_handle_t *hdl;
+
+	printf("Copy Object %lu to file %s\n", srcobj, destfile);
```

Comment:
nit : possible osx only?
zdb.c:4583:41: warning: format specifies type 'unsigned long' but the argument
      has type 'uint64_t' (aka 'unsigned long long') [-Wformat]
which is true for all your uint64_t vars in printf.